### PR TITLE
Integration tests for Organization GET API for metadata filtering

### PR DESCRIPTION
**Pull request by HasiniSama**
### Purpose
$Subject

**Changes**
- Updated **OrganizationManagementSuccessTest.java** to incorporate Success Integration tests.
- Updated **OrganizationManagementFailureTest.java** to incorporate Failure Integration tests.

### Related Issues
- https://github.com/wso2/product-is/issues/20611

**Merge After**
> https://github.com/wso2/identity-organization-management-core/pull/139
> https://github.com/wso2/identity-api-server/pull/624

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementFailureTest.java [hunk 1]
```diff
@@ -128,6 +128,24 @@ public void testSelfOnboardOrganization(String requestBodyPath) throws Exception
     }
 
     @Test(dependsOnMethods = "testSelfOnboardOrganization")
+    public void testGetOrganizationsWithInvalidOperator() {
+
+        String filterQuery = "?filter=name ca G&limit=10&recursive=false";
```
Comment (ShanChathusanda93) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementFailureTest.java:133]: `ca` can you please elaborate this?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 2]
```diff
@@ -434,6 +462,15 @@ public void testUpdateDiscoveryAttributesOfOrganization() throws IOException {
                 .body("attributes[0].values", containsInAnyOrder("xyz.com", "example.com"));
     }
 
+    @DataProvider(name = "checkDiscoveryAttributes")
```
Comment (ShanChathusanda93) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:465]: Do we need this location change?

Comment (HasiniSama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:465]: When I initially added the `checkDiscoveryAttributes` data provider, I decided to keep all data providers in one place. However, the new `filterQueries` data provider would be more readable if kept closer to the test method. Therefore, I also relocated the `checkDiscoveryAttributes` data provider near the test method for consistency and to improve overall readability.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 3]
```diff
@@ -258,7 +249,44 @@ public void testGetOrganization() {
                 .body("id", equalTo(organizationID));
     }
 
-    @Test(dependsOnMethods = "testGetOrganization")
+    @DataProvider(name = "filterQueries")
+    public Object[][] createFilterQueries() {
+        return new Object[][] {
```
Comment (AnuradhaSK) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:254]:
```suggestion
    public Object[][] createFilterQueries() {
    
        return new Object[][] {
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 4]
```diff
@@ -258,7 +249,44 @@ public void testGetOrganization() {
                 .body("id", equalTo(organizationID));
     }
 
-    @Test(dependsOnMethods = "testGetOrganization")
+    @DataProvider(name = "filterQueries")
+    public Object[][] createFilterQueries() {
```
Comment (AnuradhaSK) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:253]:
```suggestion
    public Object[][] dataProviderForFilterOrganizations() {
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 5]
```diff
@@ -258,7 +249,44 @@ public void testGetOrganization() {
...
+        return new Object[][] {
+                {"?filter=name co G&limit=10&recursive=false", false, false},
+                {"?filter=attributes.Country co S&limit=10&recursive=false", true, false},
+                {"?filter=attributes.Country eq Sri Lanka&limit=10&recursive=false", true, false},
+                {"?filter=attributes.Country eq USA&limit=10&recursive=false", false, true}
+        };
+    }
+
+    @Test(dependsOnMethods = "testGetOrganization", dataProvider = "filterQueries")
+    public void testGetOrganizations(String filterQuery, boolean expectAttributes, boolean expectEmptyList) {
```
Comment (AnuradhaSK) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:263]:
```suggestion
    public void testFilterOrganizations(String filterQuery, boolean expectAttributes, boolean expectEmptyList) {
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 6]
```diff
@@ -258,7 +249,44 @@ public void testGetOrganization() {
                 .body("id", equalTo(organizationID));
     }
 
-    @Test(dependsOnMethods = "testGetOrganization")
+    @DataProvider(name = "filterQueries")
```
[the remark this answers is not in the document]
Comment (AnuradhaSK) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:252]: Having the test method name and data provider name with similar pattern would be easier to track

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java [hunk 7]
```diff
@@ -258,7 +249,44 @@ public void testGetOrganization() {
...
+    public Object[][] createFilterQueries() {
+        return new Object[][] {
+                {"?filter=name co G&limit=10&recursive=false", false, false},
+                {"?filter=attributes.Country co S&limit=10&recursive=false", true, false},
+                {"?filter=attributes.Country eq Sri Lanka&limit=10&recursive=false", true, false},
+                {"?filter=attributes.Country eq USA&limit=10&recursive=false", false, true}
+        };
```
Comment (AnuradhaSK) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:259]: Shall we and one other case to check 
filter=attributes.Country eq USA and name co `<something>`

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java (hunk 3 above)
Comment (HasiniSama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java]: This is now updated.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java (hunk 4 above)
Comment (HasiniSama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java]: This is now updated to `dataProviderForFilterOrganizations`

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (HasiniSama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java]: This is now updated.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java (hunk 5 above)
Comment (HasiniSama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java]: This is now updated to `testFilterOrganizations`

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java (hunk 7 above)
Comment (HasiniSama) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java:261]: Added a new test case for multiple attribute filtering. 

`?filter=attributes.Country eq Sri Lanka and name co Greater`

**Review by AnuradhaSK — APPROVED** [review]

**Review by AnuradhaSK — APPROVED** [review]

**Review by AnuradhaSK — APPROVED** [review]

**Pull request merged**